geo-traits/src/geometry.rs

```diff
@@ -9,14 +9,15 @@ use super::{
 };
 
 #[allow(clippy::type_complexity)]
-pub trait GeometryTrait<'a>: Send + Sync {
-    type Point: 'a + PointTrait;
-    type LineString: 'a + LineStringTrait<'a>;
-    type Polygon: 'a + PolygonTrait<'a>;
-    type MultiPoint: 'a + MultiPointTrait<'a>;
-    type MultiLineString: 'a + MultiLineStringTrait<'a>;
-    type MultiPolygon: 'a + MultiPolygonTrait<'a>;
-    type GeometryCollection: 'a + GeometryCollectionTrait<'a>;
+pub trait GeometryTrait<'a> {
+    type T: CoordNum;
+    type Point: 'a + PointTrait<T = Self::T>;
+    type LineString: 'a + LineStringTrait<'a, T = Self::T>;
```

Comment:
This can now be:

```rust
type LineString<'a>: 'a + LineStringTrait<'a, T = Self::T>;
```

Or something like that

Reply:
What's the benefit to adding `<'a>` onto the type name as well?

Reply:
You can remove the lifetime from the trait itself, which means any references to the trait don't need to specify that lifetime. Similar to https://github.com/georust/geo/pull/908/files

Reply:
Oh wow that is super nice!

Reply:
It ended up being a lot more work than expected but I implemented that change in https://github.com/geoarrow/geoarrow-rs/pull/253 (I'm incubating and testing out trait impls there); I'll make a PR to the traits branch with those updated changed